### PR TITLE
Refactor commands to unify daemon context handling

### PR DESCRIPTION
**Pull request by Bisk1**
This refactoring is a follow-up of https://github.com/apache/airflow/pull/34931
- unified approach to daemon process creation and moved all daemon-related code to shared util daemon_utils.py.
- moved async thread creation for scheduler so that it happens after entering daemon context (will prevent possible bugs like https://github.com/apache/airflow/issues/34816 for scheduler)
- reused some utils like removing PID file of a dead process for all other commands in daemon mode

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by Bisk1** [comment]
Tested those commands manually and looks good. Unfortunately standard Python test frameworks can't handle forking processes well so no automated tests.

**Comment by Bisk1** [comment]
converting to draft becuase I still need to fix some unit tests

File: airflow/cli/commands/daemon_utils.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        Process name used in naming log and PID files for the daemon
+    callback : Callable
+        The actual command to run with or without daemon context
+    should_setup_logging : bool
+        If true, then a log file handler for the daemon process will be created
+    umask : str
+        File access creation mask ("umask") to set for the process on daemon start
+    pid_file : str
+        If specified, this file path us used to store daemon process PID.
+        If not specified, a file path is generated with the default pattern.
```
Comment (Taragolis) [inline airflow/cli/commands/daemon_utils.py]: This should be in reStructuredText docstring style

Comment (Bisk1) [reply airflow/cli/commands/daemon_utils.py]: done

File: airflow/cli/commands/celery_command.py [hunk 2]
```diff
@@ -68,28 +67,7 @@ def flower(args):
...
-                pidfile=TimeoutPIDLockFile(pidfile, -1),
-                stdout=stdout,
-                stderr=stderr,
-                umask=int(settings.DAEMON_UMASK, 8),
-            )
-            with ctx:
-                celery_app.start(options)
-    else:
-        celery_app.start(options)
+    run_command_with_daemon_mode(args, "flower", lambda: celery_app.start(options))
```
Comment (potiuk) [inline airflow/cli/commands/celery_command.py]:
```suggestion
    run_command_with_daemon_option(args, "flower", lambda: celery_app.start(options))
```

Maybe? The name suggest that it's always daemon but it is based on args.daemon .

File: airflow/cli/commands/celery_command.py [hunk 3]
```diff
@@ -214,33 +187,31 @@ def worker(args):
...
         with _serve_logs(skip_serve_logs):
             celery_app.worker_main(options)
 
+    if args.umask:
+        umask = args.umask
+    else:
+        umask = conf.get("celery", "worker_umask", fallback=settings.DAEMON_UMASK)
+
+    run_command_with_daemon_mode(
+        args,
```
Comment (potiuk) [inline airflow/cli/commands/celery_command.py]: Maybe we should use keywords-only params here?

File: airflow/cli/commands/internal_api_command.py [hunk 4]
```diff
@@ -148,60 +139,54 @@ def kill_proc(signum, _):
                 gunicorn_master_proc.kill()
             sys.exit(0)
 
-        def monitor_gunicorn(gunicorn_master_pid: int):
+        def monitor_gunicorn(gunicorn_master_proc: psutil.Process | subprocess.Popen):
```
Comment (potiuk) [inline airflow/cli/commands/internal_api_command.py:146]: I remember this psutikk/subprocess dualism hit us before. Nice to have it typed here, bit it's not typed in kill_proc - I do not know by heart if all teh methods are supported in both ? Can you add type hint there please?

**Review by potiuk — COMMENTED** [review]
Few - mostly NITs

File: airflow/cli/commands/celery_command.py (hunk 3 above)
Comment (daniel-dylag-openx) [reply airflow/cli/commands/celery_command.py]: Changed all parametry to keyword-only

File: airflow/cli/commands/celery_command.py (hunk 2 above)
Comment (Bisk1) [reply airflow/cli/commands/celery_command.py]: sure, renamed

File: airflow/cli/commands/celery_command.py (hunk 3 above)
Comment (Bisk1) [reply airflow/cli/commands/celery_command.py]: changed all params to keyword-only

File: airflow/cli/commands/internal_api_command.py (hunk 4 above)
Comment (Bisk1) [reply airflow/cli/commands/internal_api_command.py:146]: added types and looked closer at kill_proc, turns out there was some error in code related to this duality that I fixed by copying&pasting the same section from webserver_command.py (BTW these 2 files have a lot of duplication so could use more refactoring)

**Review by potiuk — APPROVED** [review]
LGTM

**Comment by potiuk** [comment]
@Taragolis ?

**Review by Taragolis — APPROVED** [review]

**Comment by potiuk** [comment]
Nice one @Bisk1 !

**Pull request merged**